docs/memory-safety-continuum.md

```diff
@@ -0,0 +1,66 @@
+# Memory Safety Continuum
```

Comment:
I wonder if this should merge into the definitions.md, since it extends upon the notions in that and that way we would have a sharable, unified artifact that defines memory-safety?

Reply:
Possibly - though I'd like to keep it in a separate doc until we further discuss and refine the idea of the continuum.